### PR TITLE
Base goerli

### PR DESCRIPTION
**Pull request by Saty248**
- [x ] Because this PR includes a **bug fix**, relevant tests have been included.
- [ ] Because this PR includes a **new feature**, the change was previously discussed on an Issue or with someone from the team.
- [ ] I didn't do anything of this.

---

i have added support for base_goerli and added relevent test for it.. @fvictorio @alcuadrado  kindly give it a watch and recommend me if any changes are there to be made . issue #3803 
i have also modified the doc to solve the issue #3812

### CODE REVIEW

File: packages/hardhat-etherscan/test/unit/resolveEtherscanApiKey.ts [hunk 1]
```diff
@@ -45,7 +45,12 @@ describe("Etherscan API Key resolution", () => {
         expectedBadApiKeyMessage
       );
     });
-
+    it("should return epmty apiKey  if api key root is empty string and network is base-goerli", () => {
```
Comment (MarioRodrigues10) [inline packages/hardhat-etherscan/test/unit/resolveEtherscanApiKey.ts:48]:
```suggestion
    it("should return empty apiKey if api key root is empty string and network is base-goerli", () => {
```

**Comment by fvictorio** [comment]
Hi @Saty248, please use separate PRs for different issues. This PR should only include the changes related to adding Base goerli.

**Comment by Saty248** [comment]
> Hi @Saty248, please use separate PRs for different issues. This PR should only include the changes related to adding Base goerli.
sure, am working on it

**Comment by schaable** [comment]
Hi @Saty248, to move things a bit faster I created [this pull request](https://github.com/NomicFoundation/hardhat/pull/4109) which adds the Base Goerli testnet. I'm going to close this one, but feel free to submit a separate PR for the Chai Matchers change.

**Pull request closed without merging**